examples/pwm_custom.rs

```diff
@@ -59,11 +35,7 @@ fn main() -> ! {
     let p1 = gpiob.pb5.into_alternate_push_pull(&mut gpiob.crl);
 
     let mut pwm = Timer::tim3(p.TIM3, &clocks, &mut rcc.apb1)
-        .pwm(
-            MyChannels(p0, p1),
-            &mut afio.mapr,
-            1.khz(),
-        );
+        .pwm((p0, p1, NoPin, NoPin), &mut afio.mapr, 1.khz());
```

Comment:
IIUC you have somehow to specify Tim3PartialRemap here.

Reply:
You are right. In some cases it's needed to specify remap:
```rust
    let mut pwm = Timer::tim3(p.TIM3, &clocks, &mut rcc.apb1)
        .pwm::<Tim3PartialRemap, _, _>((NoPin, NoPin, pb0, NoPin), &mut afio.mapr, 1.khz());
```

But not in this one.

Reply:
Could you please clarify how it is going to work in this case ? How proper remap will be figured out from the specified list of 2 pins ?

Reply:
For TIM3 CH1 and CH2 there is only one combination of pins for each remap and compiler can infer type of remap.
But if you specify only PB0 and/or PB1 (CH3 and CH4) compiler don't know what remap do you mean (`Tim3NoRemap` or `Tim3PartialRemap`).

Reply:
Ok, thanks. 

BTW, doc section in src/pwm.rs needs to be modified as well since now there is no need to use newtype pattern.

Reply:
There is one thing that worth mentioning in document section. Previous implementation returned tuple of activated channels, the type ```Channels``` was defined as a part of newtype pattern in each case. Now tuple of 4 channels will be returned in all the cases. So you have to remember which channels are not activated based on where ```NoPin``` are specified.